lib/handler/mruby.c

```diff
@@ -698,6 +698,15 @@ static void send_response(h2o_mruby_generator_t *generator, mrb_int status, mrb_
 
     /* use fiber in case we need to call #each */
     if (!mrb_nil_p(body)) {
+        /* check the body hasn't already consumed */
+        mrb_sym consume = mrb_intern_lit(mrb, "consume!");
+        if (mrb_respond_to(mrb, body, consume)) {
+            mrb_funcall_argv(mrb, body, consume, 0, NULL);
```

Comment:
@kazuho I did it in https://github.com/h2o/h2o/pull/1337/commits/551e1d606fef9c8a9551aaeb82e154612808ebdc